src/Adapters/DiagnosticSource/DiagnosticSourceListenerSubscription.cs

```diff
@@ -0,0 +1,70 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="DiagnosticSourceListenerSubscription.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.ApplicationInsights.DiagnosticSourceListener
+{
+    using Microsoft.ApplicationInsights.DataContracts;
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    internal class DiagnosticSourceListenerSubscription : IObserver<KeyValuePair<string, object>>
+    {
+        private readonly string listenerName;
+        private readonly TelemetryClient telemetryClient;
+
+        public DiagnosticSourceListenerSubscription(string listenerName, TelemetryClient telemetryClient)
+        {
+            if (listenerName == null)
+            {
+                throw new ArgumentNullException(nameof(listenerName));
+            }
+
+            if (telemetryClient == null)
+            {
+                throw new ArgumentNullException(nameof(telemetryClient));
+            }
+
+            this.listenerName = listenerName;
+            this.telemetryClient = telemetryClient;
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        public void OnError(Exception error)
```

Comment:
Consider reporting this using Event Source like this https://github.com/Microsoft/ApplicationInsights-dotnet/blob/151509eac5e23fcbd09f3a8989a35d1099703957/src/Core/Managed/Shared/Extensibility/Implementation/Tracing/CoreEventSource.cs#L10